File: Remove-Spaces-from-a-given-String/sol.cpp
class Solution {
public:
    string reverseWords(string s) {
        int l =s.size();
        string sAns="";
        string temp="";
        for(int i=0;s[i]!='\0';i++){
            if(s[i]==' '){
                if(temp != ""){
                    if(sAns=="")
                        sAns = temp;
                    else
                        sAns = temp + " " + sAns;
                    temp="";
                }
                continue;
            }
            temp+=s[i];
        }
        if(temp!="")
            sAns = temp + " " + sAns;
        s.erase(remove(s.begin(), s.end(), ' '), s.end());
        if(temp==s)
            return temp;
        return sAns;
    }
};

File: Permmutations/sol.java
class Solution {

    public void permutations(int[] nums, int start, List<List<Integer>> sol, List<Integer> arr){
        if(start==nums.length){
            sol.add(new ArrayList<>(arr));
            return;
        }

        for(int i=start;i<nums.length;i++){        
            int temp = nums[i];
            nums[i] = nums[start];
            nums[start] = temp;
            arr.add(temp);
            permutations(nums, start+1, sol, arr);
            nums[start] = nums[i];
            nums[i] = temp;
            arr.remove(arr.size()-1);
        }
    }

    public List<List<Integer>> permute(int[] nums) {
        List<List<Integer>> sol = new ArrayList<>();
        permutations(nums, 0, sol, new ArrayList<>());
        return sol;
    }
}
